mono/mini/simd-intrinsics-netcore.c

```diff
@@ -309,12 +309,20 @@ emit_sys_numerics_vector_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSig
 		return emit_xcompare (cfg, klass, etype, ins, ins);
 	}
 	case SN_get_Item:
-		if (args [1]->opcode != OP_ICONST)
+		MONO_EMIT_NEW_BIALU_IMM (cfg, OP_COMPARE_IMM, -1, args [1]->dreg, len);
+		MONO_EMIT_NEW_COND_EXC (cfg, GE_UN, "IndexOutOfRangeException");
+		if (etype->type == MONO_TYPE_R4 || etype->type == MONO_TYPE_R8)
+			// FIXME:
```

Comment:
Note: The float version is used by [this benchmark](https://github.com/dotnet/performance/blob/4c3f3f8ac7e160b2c3606a6da7e341b9f82505a5/src/benchmarks/micro/coreclr/SIMD/ConsoleMandel/VectorFloat.cs#L85).